### ISSUE TITLE
Re-think the from_array method

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/GeoUtils#115]
Currently, child classes of Raster also inherit the from_array function, which always returns a Raster instance. This is problematic for example with reproject, which returns a Raster instance:
Here's a minimum working example:
```
from geoutils import georaster as gr
from geoutils import datasets
from geoutils import satimg
img = satimg.SatelliteImage(datasets.get_path('landsat_B4'))
img2 = satimg.SatelliteImage(datasets.get_path('landsat_B4_crop'))
test = img.reproject(dst_ref=img2)
type(test)
```

**Comment by adehecq** [comment]
Actually, from_array works as expected,a s it returns an object of the right type, see below. The problem of reproject must be elsewhere...
```
img_out = img.from_array(img.data, transform=img.transform, crs=img.crs)
type(img_out) == type(img)
```

**Comment by adehecq** [comment]
We also need to add tests for this case, when solving the issue.

**Comment by erikmannerfelt** [comment]
Maybe link this to #116 ?

**Comment by adehecq** [comment]
Fixed by #116
